ccloader/js/filemanager.js

```diff
@@ -42,30 +50,43 @@ export class Filemanager {
 		return this._loadScript(file, document, isModule ? 'module' : 'text/javascript');
 	}
 
-	/**
-	 *
-	 * @param {string} folder
-	 */
-	getAllModsFiles(folder) {
-		const subs = this._getFolders(folder);
-		return [].concat(...subs.map(sub => this._getResourcesInFolder(sub, path.sep + 'package.json')));
+	getAllModsFiles() {
+		const subs = this._getModFoldersOnAndroid('/') || this._getFolders();
+		return [].concat(...subs.map(sub => this._getResourcesInFolder(sub, this.pathSep + 'package.json')));
 	}
 
-	/**
-	 *
-	 * @param {string} folder
-	 */
-	getAllCCModFiles(folder) {
-		const subs = this._getFolders(folder);
-		return [].concat(...subs.map(sub => this._getResourcesInFolder(sub, path.sep + 'ccmod.json')));
+	getAllCCModFiles() {
+		const subs = this._getModFoldersOnAndroid('/') || this._getFolders();
+		return [].concat(...subs.map(sub => this._getResourcesInFolder(sub, this.pathSep + 'ccmod.json')));
 	}
 
-	/**
-	 *
-	 * @param {string} folder
-	 */
-	getAllModPackages(folder) {
-		return this._getResourcesInFolder(folder, '.ccmod');
+	getAllModPackages() {
+		return this._getModFoldersOnAndroid('.ccmod') || this._getResourcesInFolder(null, '.ccmod');
+	}
+
+	_getModFoldersOnAndroid(ending) {
+		const provider = window.CrossAndroidModListProvider;
+		if (provider && provider.getModListAsJson) {
+			const entries = [];
+			for (let entry of JSON.parse(provider.getModListAsJson())) {
+				if (entry.endsWith(ending)) {
+					if (ending === '/') {
+						entry = entry.slice(0, -1);
+					}
+					entries.push('assets/mods/' + entry);
+				}
+			}
+			return entries;
+		}
+		return null;
+	}

```

Comment:
This function has some issues:
* The content of the function looks pretty weird:
  * It's responsible for checking the architecture, returning null if it isn't the correct one. The other functions don't do that.
  * Why would `getModListAsJson` return a string and not simply the array? If it's a technical limitation wouldn't it be better to fake the `mods.json` file and use browser logic to load the files?
  * The purpose of removing the `'/'` at the end is not obvious. Please add a comment.
* Should be called by `_getFolders` and `_getResourcesInFolder` instead of `getAllModsFiles` and `getAllCCModFiles`
* Should be moved down to the other private functions
* Missing JSDoc (types + description)